src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -27,23 +27,30 @@
 import java.lang.reflect.*;
 import java.nio.CharBuffer;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletRequest;
 import java.util.Iterator;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.List;
 
 public class SugarWebAgent
   implements WebAgent
 {
+  protected Method cur_method;
+
   public SugarWebAgent() {}
 
   public void execute(X x) {
     Logger              logger         = (Logger) x.get("logger");
     PrintWriter         out            = x.get(PrintWriter.class);
     HttpServletResponse resp           = x.get(HttpServletResponse.class);
-    HttpParameters      p              = x.get(HttpParameters.class);
+    HttpServletRequest  req            = x.get(HttpServletRequest.class);
```

Comment:
Why are you not using HttpParameters?  Do we need to add map support to it?
The HttpParameters abstracts a number of parameter passing methods.